### PR TITLE
blog post announce for lts-21 and ghc-9.6 nightly

### PR DESCRIPTION
**Pull request by juhp**
Please review

### CODE REVIEW

File: posts/announce-lts-21-nightly-ghc9.6.md [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+If your package is missing from LTS 21 and builds there, you can easily request to have it added using our straightforward process: just open a github issue in the [lts-haskell](https://github.com/commercialhaskell/lts-haskell/issues) project and following the simple steps in the template.
+
+## Stackage Nightly updated to ghc-9.6.2
+
+At the same time we are also excited to have moved [Stackage Nightly](https://www.stackage.org/nightly) already to GHC 9.6.2: the initial snapshot being [nightly-2023-06-23](https://www.stackage.org/nightly-2023-06-23)!  Current nightly has almost 2449 packages, but we expect that number to continue to grow over the coming days, weeks, and months: we very much welcome your contributions and help with that.
+
+The main reasons for the having to disable packages that are now in lts-21 being changes to exports for monad functions and optparse-applicative prettyprinting, and also the openFD API change. Thanks to all those who have already done work updating their packages to ghc-9.6.
```
Comment (ysangkok) [inline posts/announce-lts-21-nightly-ghc9.6.md:20]: Very convoluted sentence, should probably be split up.  What about something like:

> Common reasons for disabling packages were:
> * unix-2.8 changed the API of `openFd`
> * mtl-2.3 stopped reporting from e.g. `Control.Monad`
> * optparse-applicative changed API

File: posts/announce-lts-21-nightly-ghc9.6.md [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+timestamp: 2023-06-24T20:00:00Z
+---
+
+## Stackage LTS 21 has been released
+
+The Stackage team is very happy to announce the first [Stackage LTS version 21](https://www.stackage.org/lts-21.0) snapshot has been released this week, based on GHC stable version 9.4.5.
+
+LTS 21 includes many [package changes](https://www.stackage.org/diff/lts-20.26/lts-21.0), and again has over 3000 packages!
+Thank you for all the nightly contributions that made this possible.
+(By the way lts-20.0 is basically identical to [nightly-2023-06-20](https://www.stackage.org/nightly-2023-06-20).)
```
Comment (alaendle) [inline posts/announce-lts-21-nightly-ghc9.6.md:14]: lts-20.0 - should be 21?

**Review by alaendle — APPROVED** [review]

File: posts/announce-lts-21-nightly-ghc9.6.md (hunk 2 above)
Comment (juhp) [reply posts/announce-lts-21-nightly-ghc9.6.md]: Good catch

**Comment by alaendle** [comment]
Many thanks to @juhp for driving things forward in such an exemplary manner.

File: posts/announce-lts-21-nightly-ghc9.6.md [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+LTS 21 includes many [package changes](https://www.stackage.org/diff/lts-20.26/lts-21.0), and again has over 3000 packages!
+Thank you for all the nightly contributions that made this possible.
+(By the way lts-21.0 is basically identical to [nightly-2023-06-20](https://www.stackage.org/nightly-2023-06-20).)
+
+If your package is missing from LTS 21 and builds there, you can easily request to have it added using our straightforward process: just open a github issue in the [lts-haskell](https://github.com/commercialhaskell/lts-haskell/issues) project and following the simple steps in the template.
+
+## Stackage Nightly updated to ghc-9.6.2
+
+At the same time we are also excited to have moved [Stackage Nightly](https://www.stackage.org/nightly) already to GHC 9.6.2: the initial snapshot being [nightly-2023-06-22](https://www.stackage.org/nightly-2023-06-22)!  Current nightly has 2449 packages, but we expect that number to continue to grow over the coming days, weeks, and months: we very much welcome your contributions and help with that.
```
Comment (mihaimaruseac) [inline posts/announce-lts-21-nightly-ghc9.6.md:20]:
```suggestion
At the same time we are also excited to have moved [Stackage Nightly](https://www.stackage.org/nightly) to GHC 9.6.2: the initial snapshot being [nightly-2023-06-22](https://www.stackage.org/nightly-2023-06-22).  Current nightly has 2449 packages, but we expect that number to continue to grow over the coming days, weeks, and months: we very much welcome your contributions and help with that.
```

File: posts/announce-lts-21-nightly-ghc9.6.md [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+
+- mtl-2.3 no longer re-exporting from Control.Monad etc
+- optparse-applicative-0.18 changes to help prettyprinting exports
+- unix-2.8 changing the API of openFd.
+
+Thank you to all those who have already done work updating their packages to ghc-9.6.
+
+Adding or enabling your package for Nightly is just a simple [pull request](https://github.com/commercialhaskell/stackage/blob/master/MAINTAINERS.md#adding-a-package) to a large yaml file.
+
+Happy stackage building
```
Comment (mihaimaruseac) [inline posts/announce-lts-21-nightly-ghc9.6.md:32]:
```suggestion
Happy stackage building!
```

**Review by mihaimaruseac — APPROVED** [review]

**Comment by juhp** [comment]
And thanks to @alaendle Andreas for pushing them over the line swiftly!

**Review by snoyberg — APPROVED** [review]

**Comment by juhp** [comment]
Thank you everyone!

**Pull request merged**